frontend/src/Components/Customers/Row.js

```diff
@@ -0,0 +1,94 @@
+import { Cancel } from '../Common/CancelButton';
+import { Edit } from '../Common/Edit';
+import { useState } from 'react';
+
+export function Row(props) {
+    const [isEditMode, setEditMode] = useState(false)
+    const [updatedCustomer, setUpdatedCustomer] = useState(props.customer)
+
+    function onPropertyChange(property, entity, updatedValue) {
+        setUpdatedCustomer({ ...entity, [property]: updatedValue })
+    }
+
+    function save(customer) {
+        axios
+            .put('/customers/' + customer.id,
+                {
+                    changes: {
+                        first_name: customer.first_name,
+                        last_name: customer.last_name,
+                        company: customer.company
+                    }
+                })
+            .then(cust => {
+                console.log(cust.data)
+                setEditMode(!isEditMode)
+            })
+    }
+
+    function onClick() {
+        setEditMode(true)
+        props.onClick()
+    }
+
+    function onCancelClick() {
+        setEditMode(false)
+    }
+
+    return (
+        <tr>
+            <th scope="row">{updatedCustomer.id}</th>
+            <td>
+                {
+                    isEditMode && props.selectedRowIndex === updatedCustomer.id ?
+                        <Edit value={updatedCustomer.first_name}
+                            id={'first_name'}
+                            onChangeHandler={(newValue) => onPropertyChange('first_name', updatedCustomer, newValue)} /> :
+                        updatedCustomer.first_name
+                }
+            </td>
+            <td>
+                {
+                    isEditMode && props.selectedRowIndex === updatedCustomer.id ?
+                        <Edit value={updatedCustomer.last_name}
+                            id={'last_name'}
+                            onChangeHandler={(newValue) => onPropertyChange('last_name', updatedCustomer, newValue)} /> :
+                        updatedCustomer.last_name
+                }
+            </td>
+            <td>
+                {
+                    isEditMode && props.selectedRowIndex === updatedCustomer.id ?
+                        <Edit value={updatedCustomer.company}
+                            id={'company'}
+                            onChangeHandler={(newValue) => onPropertyChange('company', updatedCustomer, newValue)} /> :
+                        updatedCustomer.company
+                }
+            </td>
+            <td>
+                {
+                    isEditMode && props.selectedRowIndex === updatedCustomer.id ?
+                        <div className="btn-group" role="group">
+                            <SaveButton saveHandler={() => save(updatedCustomer)} />
+                            <Cancel onClick={onCancelClick}>
+                                <svg width="1em" height="1em" viewBox="0 0 16 16" className="bi bi-x-circle" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
```

Comment:
As fi pus aceste SVGs intr-un fisier separat de assets, sa tina codul scurt si curat. Merg importate fisiere SVG in React.